ci/test-checks.sh

```diff
@@ -76,6 +76,7 @@ _ scripts/cargo-for-all-lock-files.sh -- "+${rust_nightly}" clippy --workspace -
   --deny=clippy::integer_arithmetic \
   --deny=clippy::manual_let_else \
   --deny=clippy::used_underscore_binding \
+  --allow=clippy::items-after-test-module \
```

Comment:
https://github.com/rust-lang/rust-clippy/issues/11153

allowing this one globally Feels Bad Man :tm:, but the offending code is emitted by a proc-macro and we can't put it in the right place locally :face_with_head_bandage: